src/OIGenerator.cpp

```diff
@@ -18,40 +18,77 @@
 
 #include <glog/logging.h>
 
+#include <boost/core/demangle.hpp>
 #include <fstream>
+#include <unordered_map>
 #include <variant>
 
 #include "DrgnUtils.h"
 #include "OIUtils.h"
 
 namespace ObjectIntrospection {
 
+std::unordered_map<std::string, std::string>
+OIGenerator::oilStrongToWeakSymbolsMap(drgnplusplus::program& prog) {
+  static constexpr std::string_view strongSymbolPrefix =
+      "int ObjectIntrospection::getObjectSize<";
+  static constexpr std::string_view weakSymbolPrefix =
+      "int ObjectIntrospection::getObjectSizeImpl<";
+
+  std::unordered_map<std::string, std::pair<std::string, std::string>>
+      templateArgsToSymbolsMap;
+
+  auto symbols = prog.find_symbols_by_name(nullptr);
+  for (drgn_symbol* sym : *symbols) {
```

Comment:
I would just double check this dosen't slow things down too much. I don't think it should though!

Reply:
In terms of the dereference? Looked at the ASM and it's fine. If it's about whether that drgn function loads all the DWARF, that's a problem I'm happy to push down the road for now - this program is slow until I add the new drgn feature anyway. Thanks for the check!